### DISCUSSION TITLE
POCSAG - Modulation Data and PTT output - High Power Application

### DISCUSSION THREAD

**Discussion by mkrebbers** [jgromes/RadioLib#927] [Ideas]
After working with Radiolib and C1101 Pocsag was a success i am starting the next Project. For my Ericsson Compact 9000 i would like to use Radiolib to send Modulation Data/Clock and also PTT directly to the bus. Is there any Idea on getting the raw modulation data acc to the Pocsag standard out of radiolib and trigger the PTT?

**Comment by jgromes** [comment]
What is suposed to be the interface between RadioLib and the tranmsitter?

During transmission, RadioLib uses direct mode transmission, which basically just commands the module to transmit at a variable frequency, based on whether we need to transmit a 1 or a 0.

There's also the `ExternalRadio` class which allows you to output `PagerClient` transmission on a digital pin. High frequency will be turned into digital 1, low frequency into digital 0.

**Reply by mkrebbers** [reply]
The Interface would be as following: 

MDL (Logic Level): Modulation Data Bitstream to C9000 
MDE (Logic Level): Modulation Enable to C9000 has to be low on every rising/falling edge of MDL Bitstream
PTT (Logic Level): Transmitter PTT to C9000 enables the Transmitter (Transmitter key delay aprox. 20ms)

Would it be an idea for you to add the C9000 as seperate Radio? 

In second instance i would like to add the remote Power control of the Transmiter which would be 2-8V Analog signal via ADC and also a readout of the C9000 VWSR parameters "Forward" and "Reverse" into the ESP32 to check the output. Could be external Code or maybe Power Amplifier Class. 

For the Power Amplifier this would mean: 
Power (Analog): 2-8V ADC Out to C9000 sets Power from 25-210W 
FWD (Analog): ADC Input not yet defined
REF (Analog): ADC Input not yet defined

**Reply by jgromes** [reply]
> Would it be an idea for you to add the C9000 as seperate Radio?

That falls outside of the library scope, it's intended as a driver for transceiver ICs, not ham radio rigs. This is what ExternalRadio is for.

Regarding the input to your radio, like I wrote before - RadioLib has the option to output the data using the ExternalRadio class. I don't know if it's going to work in your specific setup, you just have to try.

> In second instance i would like to add the remote Power control of the Transmiter which would be 2-8V Analog signal via ADC and also a readout of the C9000 VWSR parameters

Again, that's up to you, since it's really specific to your setup. Definitely out of scope for a generic library.

**Reply by ManoDaSilva** [reply]
This might help, we had a discussion on the subject... Being tested by @fred-corp out as we speak! :)

https://github.com/jgromes/RadioLib/discussions/646
